tests/study_tests/test_multi_objective.py

```diff
@@ -110,3 +112,41 @@ def test_dominates_complete_vs_incomplete(t1_state: TrialState) -> None:
 
     assert _dominates(t2, t1, list(directions))
     assert not _dominates(t1, t2, list(directions))
+
+
+def test_calculate_nondomination_rank() -> None:
```

Comment:
Tests for duplicated trials are missing, so I think it is better adding them.
```python
test_case = np.array([[1, 1], [1, 1], [1, 2], [2, 1], [0, 1.5], [1.5, 0], [0, 1.5]])
ans = [0, 0, 1, 1, 0, 0, 0]

test_case = np.array([[1, 1], [1, 1], [1, 2], [2, 1], [1, 1], [0, 1.5], [0, 1.5]])
ans = [0, 0, 1, 1, 0, 0, 0]
```

Also, could you add something similar for the case of `n_objectives=3`?